pkg/providers/alibaba/template.go

```diff
@@ -50,195 +49,178 @@ rules:
-      tier: control-plane
   template:
     metadata:
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ""
       labels:
         app: "alibaba-ccm"
-        tier: control-plane
-      annotations:
-        scheduler.alpha.kubernetes.io/critical-pod: ''
     spec:
-      serviceAccountName: cloud-controller-manager
-      tolerations:
-        - effect: NoSchedule
-          operator: Exists
-          key: node-role.kubernetes.io/master
-        - effect: NoSchedule
-          operator: Exists
-          key: node.cloudprovider.kubernetes.io/uninitialized
-      nodeSelector:
-        node-role.kubernetes.io/master: "true"
       containers:
         - command:
-            -  /cloud-controller-manager
+            - /cloud-controller-manager
             - --kubeconfig=/etc/kubernetes/k3s.yaml
-            - --address=127.0.0.1
-            - --allow-untagged-cloud=true
-            - --leader-elect=true
-            - --cloud-provider=alicloud
+            - --cloud-config=/etc/kubernetes/config/cloud-config.conf
+            - --metrics-bind-addr=0
+            - --configure-cloud-routes=false
             - --allocate-node-cidrs=true
             - --cluster-cidr=%s
-            - --use-service-account-credentials=true
-            - --route-reconciliation-period=30s
-            - --v=5
-          image: registry.%s.aliyuncs.com/acs/cloud-controller-manager-amd64:v1.9.3.239-g40d97e1-aliyun
-          env:
-            - name: ACCESS_KEY_ID
-              valueFrom:
-                configMapKeyRef:
-                  name: cloud-config
-                  key: special.keyid
-            - name: ACCESS_KEY_SECRET
-              valueFrom:
-                configMapKeyRef:
-                  name: cloud-config
+          image: registry.%s.aliyuncs.com/acs/cloud-controller-manager-amd64:v2.4.0
+          imagePullPolicy: IfNotPresent
+          env:  
+            - name: ACCESS_KEY_ID 
+              valueFrom:  
+                configMapKeyRef:  
+                  name: cloud-config  
+                  key: special.keyid  
+            - name: ACCESS_KEY_SECRET 
+              valueFrom:  
+                configMapKeyRef:  
+                  name: cloud-config  
                   key: special.keysecret
           livenessProbe:
             failureThreshold: 8
             httpGet:
               host: 127.0.0.1
               path: /healthz
-              port: 10252
+              port: 10258
               scheme: HTTP
             initialDelaySeconds: 15
+            periodSeconds: 10
+            successThreshold: 1
             timeoutSeconds: 15
           name: cloud-controller-manager
           resources:
+            limits:
+              cpu: "1"
+              memory: 1Gi
             requests:
-              cpu: 200m
+              cpu: 100m
+              memory: 200Mi
           volumeMounts:
-            - mountPath: /etc/kubernetes/
+            - mountPath: /etc/kubernetes/k3s.yaml
```

Comment:
It is very dangerous to mount the admin kubeconfig. Is it any workaround for this?

Reply:
The file's perm is `readOnly` &  CCM needs `kubeconfig` file.